src/libsystemd-network/network-internal.c

```diff
@@ -509,3 +509,31 @@ int deserialize_dhcp_routes(struct sd_dhcp_route **ret, size_t *ret_size, size_t
 
         return 0;
 }
+
+int serialize_dhcp_option(FILE *f, const char *key, const uint8_t *data, size_t size) {
+        _cleanup_free_ char *hex_buf = NULL;
+
+        assert(f);
+        assert(key);
+        assert(data);
+        assert(size);
+
+        hex_buf = hexmem(data, size);
+        if (hex_buf == NULL)
+                return -ENOMEM;
+
+        fprintf(f, "%s=%s\n", key, hex_buf);
+
+        return 0;
+}
+
+int deserialize_dhcp_option(uint8_t **data, size_t *data_len, const char *string) {
+        assert(data);
+        assert(data_len);
+        assert(string);
+
+        if (strlen(string) % 2)
```

Comment:
unhexmem just rounds up to make sure that there is enough space. This check is just validating that this is a hex string (which will always be an even length). The client_id and vendor_specific deserialization were doing the same check.